libs/base/ui/form/multi-range/src/multi-range.component.ts

```diff
@@ -13,142 +12,123 @@ export class MultiRangeComponent
-  }
-
-  update(changedProperties: PropertyValues): void {
-    const calculatePercentage = (
-      value: number,
-      minValue: number,
-      maxValue: number
-    ) => ((value - minValue) / (maxValue - minValue)) * 100;
-    const sliderMinPercentage = calculatePercentage(
-      this.minValue,
-      this.min,
-      this.max
-    );
-    const sliderMaxPercentage = calculatePercentage(
-      this.maxValue,
-      this.min,
-      this.max
-    );
     this.style.setProperty('--_multi-range-min', `${sliderMinPercentage}%`);
     this.style.setProperty(
       '--_multi-range-max',
       `${100 - sliderMaxPercentage}%`
     );
+  }
+
+  protected syncNativeInputValues(minValue: number, maxValue: number): void {
+    this._min = minValue;
+    this._max = maxValue;
 
-    super.update(changedProperties);
+    if (this.inputMin) {
+      this.inputMin.value = String(minValue);
+    }
+    if (this.inputMax) {
+      this.inputMax.value = String(maxValue);
+    }
   }
 
-  updated(): void {
-    this.syncNativeInputValues();
+  protected ensureValues(
+    properties: PropertyValues<MultiRangeProperties>
+  ): void {
+    if (this.hasInvalidRange() || !this.hasDiffs(properties)) return;
 
-    this.dispatchEvent(
-      new CustomEvent<MultiRangeChangeEvent>('change', {
-        bubbles: true,
-        composed: true,
-        detail: {
-          minValue: this.minValue,
-          maxValue: this.maxValue,
-        },
-      })
-    );
+    let minValue = this.minValue!;
+    let maxValue = this.maxValue!;
+    const min = this.min!;
+    const max = this.max!;
+
+    const minInRange = minValue >= min && minValue <= max - this.step;
+    const maxInRange = maxValue >= min + this.step && maxValue <= max;
+    const minBeforeMax = minValue <= maxValue - this.step;
+
+    if (!minInRange || !minBeforeMax) minValue = min;
+    if (!maxInRange || !minBeforeMax) maxValue = max;
+
+    this.minValue = minValue;
+    this.maxValue = maxValue;
+
+    this.setPercentages(minValue, maxValue);
+    this.syncNativeInputValues(minValue, maxValue);
   }
 
-  protected syncNativeInputValues(): void {
-    this.minValue = Number(this.inputMinRef?.value?.value ?? this.minValue);
-    this.maxValue = Number(this.inputMaxRef?.value?.value ?? this.maxValue);
+  protected hasDiffs(
+    properties: PropertyValues<MultiRangeProperties>
+  ): boolean {
+    return Array.from(properties.entries()).some(([_key, value]) => {
+      const key = _key as keyof MultiRangeComponent;
+      return this.observedKeys.includes(key) && this[key] !== value;
+    });
   }
 
-  protected override render(): TemplateResult {
+  protected hasInvalidRange(): boolean {
+    this.invalid =
+      this.observedKeys.some((key) => typeof this[key] === 'undefined') ||
+      this.min! > this.max! - this.step ||
+      this.max! < this.min! + this.step;
+
+    return this.invalid;
+  }
+
+  protected override render(): TemplateResult | void {
+    if (this.invalid) return;
```

Comment:
This is a change of behavior, so we can consider it as a fix... or wrap with feature flag. In first case we will need to change the type of commit to `fix`.

Reply:
Refactor is fine in my pov, as this component has not been used so far.